### ISSUE TITLE
custom button card how to get device attributes

### ISSUE THREAD

**Issue by jmfargas** [custom-cards/button-card#787]
Is it possible to get the device attributes. I especially need to get the device name, since it is sometimes different from the entity name, for example for devices with several entities, like the shelly Pro 4PM. The method to get it is: 
   device_attr('entity.entity_id', 'name')
In addition it would be nice to be able to get the IP address of the device, which is done by doing:
   device_attr('entity.entity_id', 'configuration_url')
Thank you very much.

**Comment by dcapslock** [comment]
Use `entity.attributes.name` and `entity.attributes.configuration_url` as javascript template variables.

**Comment by jmfargas** [comment]
The  problem is that in the custom button card entity.attributes.name and
entity.attributes.configuration_url are undefined.
Though I can get, entity.attributes.friendly_name.
Thanks,

**Comment by dcapslock** [comment]
Sorry, I note you wish for the attributes of device. You can lookup by device_id of entity. So something like. `hass.devices[entity.device_id].configuration_url` and `hass.devices[entity.device_id].name`

**Comment by jmfargas** [comment]
I'm not succeeding in making it work in my custom-button-card?
I get 'hass' is undefined in the develpoper tools template, although I'm not sure whether it will work there, since it's jinja2, right?
That's my code in template:
{{ hass.devices[sensor.rebost_arreu_temperature.device_id].name }}
Many thanks!

**Comment by dcapslock** [comment]
No jinja in button-card. All javascript. `[[[  ]]]`. Check the docs.

**Comment by dcapslock** [comment]
Post your config and I will mark up where you are needing to use javascript and not jinja.

**Comment by jmfargas** [comment]
This is what I have added:
msg = " name = " + hass.devices[sensor.rebost_arreu_temperature.device_id].name;

Many thanks!

**Comment by dcapslock** [comment]
That does not help me guide you. What is you button card config? Post using proper formatting.

**Comment by jmfargas** [comment]
OK, sorry,
I have made a short example with the label of the card:
````
        label: |
          [[[
            var msg = "";
            var ent = "sensor.rebost_arreu_temperature";
            //msg = "temp: " + states[ent].state + "º";
            msg = hass.devices[ent.device_id].name;
            return msg;
          ]]]
````
In the above example, the code is correct to show the temperature in the label (it is commented), but it fails for the name.
I'm not sure if I'm calling it correctly...
Many thanks !

**Comment by dcapslock** [comment]
`ent` is a string and will not have a `device_id`. 

Try

```yaml
        label: |
          [[[
            var msg = "";
            var ent = "sensor.rebost_arreu_temperature";
            //msg = "temp: " + states[ent].state + "º";
            msg = hass.devices[hass.entities[ent].device_id].name;
            return msg;
          ]]]
```

To take the guess work out of it, look in browser developer tools, find the <button-card> element, look for the `_hass` property in properties, this is the same as `hass` available in JS templates.

**Comment by jmfargas** [comment]
WOW! Amazing, it works for both, name and configuration_url. Thanks so much!

**Comment by jmfargas** [comment]
I'll try to get the _hass property as you suggested. I guess it will show all the attributes?
Many thanks again!

**Comment by dcapslock** [comment]
Yes, when I am helping you I am always referring to that so I am not guessing.
